### ISSUE TITLE
[Documentation]: Guidance on how to use epoch definitions

### ISSUE THREAD

**Issue by TomDonoghue** [NeurodataWithoutBorders/pynwb#1450]
### What would you like changed or added to the documentation and why?

This is perhaps something of a combination of a documentation suggestions, and maybe something of a "how do I do this" question - but one that I think should probably be noted in the documentation, and that I haven't found any clear answers to. 

Specifically, I'm trying to use the `epochs` information, but having trouble figuring out what the suggested way to actually use this information when reading and analyzing the data from NWB files. If, for example, I add an epoch with some tag of interest (or, similarly, add an epoch column for "description" and add a label), then I have all the information - but subsequently reading and using this info seems like it could be easier, in a way that makes me wonder if I'm missing something. I can, for example, get the vector of tags, search it for indices that match the desired tag, and then use this to select epoch start/stop times, and subset from there - but I guess I was thinking there could / should be an easier way to select time ranges from epochs given the tags (or some added column), given that this is presumably the main goal and use case of having the epochs information. 

Relatedly, in some cases `trials` actually seems to support this better. For example, some of the "epoch" information I have is per trial (we have long trials with different phases, like encoding & recall), and so far I had been using trial information, adding, for example, `encoding_start` and `encoding_end` as trial columns, which actually allows for much easier subsetting (since `nwbfile.trials.encoding_start[:]` and similar for stop directly gives me the timestamps I want). However, given this information is time ranges, when looking into epochs, it seemed to me that this should be the preferred place to store this information, but doing so seems to make reading & using these timestamps less direct.

More specifically, the usage questions are basically:
- Is there a recommended approach for using epoch definitions to easily get timestamp values to use to extract data? I'm not sure if I'm missing something for selecting epochs and/or for directly sub-setting data fields using interval definitions. 
- Maybe I'm mis-understanding the goal of `epochs`. For time-range data that exist within a "trial" structure is the recommendation to use `epochs` or start / stop time in trials? 
    - Perhaps trials is generally preferred here, which avoids some of my confusion, though I think if so some documentation notes and what should go where would be useful, and I think the above question(s) are still relevant to something like defining and reading "rest epochs" or similar. 

In terms of documentation suggestions, I think that it could be useful if there were more examples of recommended approaches for how to read & use NWB format data. So far, I couldn't find any examples that show using trial and/or epoch information (other than simply loading them), and that's part of why I'm really not sure if I'm just missing some functionality, or not understanding what information is supposed to go where and how to use it. Potentially new / updated examples could, for example, briefly show specific examples of using trial & epoch data (sub-setting data streams based on timestamps). I think this could be useful in terms of "showing off" how generally useful NWB files are, and clarifying recommended approaches. 

In general, if this does lead to some additions to the documentations, I would be happy to help add this - though I would at least initially need some guidance on what the recommended strategies are. 

### Do you have any interest in helping write or edit the documentation?

Yes, but I would need guidance.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you checked the [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this change was not already [requested](https://github.com/NeurodataWithoutBorders/pynwb/issues)?

**Comment by bendichter** [comment]
@TomDonoghue it seems like what you want to do is something like:

 ```python
from pynwb import NWBFile
import datetime

nwbfile = NWBFile('aa', 'aa', datetime.datetime.now())

nwbfile.add_epoch(start_time=1., stop_time=3., tags=('train', 'test', 'stim'))
nwbfile.add_epoch(start_time=3., stop_time=5., tags=('test', 'stim'))
nwbfile.add_epoch(start_time=3., stop_time=5., tags=('test', 'train'))

inds = np.where(['train' in x for x in nwbfile.epochs['tags'][:]])[0]

nwbfile.epochs['start_time'][inds]
```

is that right?

**Comment by TomDonoghue** [comment]
@bendichter - yeh, that approach would definitely work - I guess I was mostly wondering if that's the expected way to use epoch definitions, since it doesn't feel like that is the easiest way to use this info. 

By comparison, at least for my use cases, I find either of the following are potentially easier to use:

### Using Trial Info
Note: this only works for intervals defined within the overall trial structure
```
from pynwb import NWBFile
import datetime

nwbfile = NWBFile('aa', 'aa', datetime.datetime.now())
nwbfile.add_trial_column('interval_name_start', 'XX')
nwbfile.add_trial_column('interval_name_stop', 'XX')
nwbfile.add_trial(start_time=0., stop_time=5., interval_name_start=2., interval_name_stop=3.)
nwbfile.add_trial(start_time=10., stop_time=15., interval_name_start=12., interval_name_stop=13.)

# Access start times
nwbfile.trials['interval_name_start'][:]
```

### Using Custom Intervals
Note: this generalizes, but is potentially annoying / sub-optimal in terms of defining a new TimeIntervals object for each interval label (which will be disconnected in the file). 
```
# CUSTOM INTERVALS
from pynwb import NWBFile
from pynwb.epoch import TimeIntervals
import datetime

nwbfile = NWBFile('aa', 'aa', datetime.datetime.now())

intervals = TimeIntervals(name="interval_name", description="XX")
intervals.add_row(start_time=2., stop_time=3.)
intervals.add_row(start_time=12., stop_time=13.)
nwbfile.add_time_intervals(intervals)

# Access start times
nwbfile.intervals['interval_name']['start_time'][:]
```

Basically, I find it nicer / easier to be able to select interval range by some kind of name, which can be done in either of the above with `interval_name`, but isn't clearly supported to do like this when using epochs. I thought using `epochs` might be the recommended way to store the interval data I have, but I wasn't clear on what would be the expected / suggested usage of `epochs` in particular, and though the above approaches seem to work well, I didn't / don't want to wander too far from the recommended / expected approach in terms of standardization.

**Comment by bendichter** [comment]
Epochs are for stages of an experiment e.g. training vs. testing. For trials-like times I'd use trials or another custom TimeIntervals table
